File: javascript/secondLargest.js
/**
 * @param {string} s
 * @return {number}
 */
var secondHighest = function (s) {

    let largest = -1
    let secondLargest = -1

    for (let l = 0; l <= s.length ; l++) {
        if (s[l] >= '0' && s[l] <= '9') {
            let digit = parseInt(s[l]);

            if (digit > largest) {
                secondLargest = largest;
                largest = digit;
            }
            else if (digit > secondLargest && digit != largest) {
                secondLargest = digit;
            }
        }
    }
    return secondLargest;
};

File: c++/string/secondLargest.cpp
class Solution {
public:
    int secondHighest(string s) {
        
        int ans = -1;
        int largest = -1;

        for(int i = 0 ; i < s.size() ; i++){
            if(s[i] >='0' && s[i] <='9' ){

                int digit = s[i] - '0' ;


                if(digit > largest){
                    ans = largest ; 
                    largest = digit ;
                }
                else if(digit > ans && digit != largest ){
                    ans = digit ;
                }
            }

        }
        return ans ;
        
    }
};